catalog/spatial/csw/spatial-csw-transformer/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/TestGmdTransformer.java

```diff
@@ -0,0 +1,140 @@
+            "elevation",
+            "inlandWaters",
+            "oceans",
+            "society",
+            "structure",
+            "transportation",
+            "utilitiesCommunication"));
+
+    private void assertGmdMetacard(Metacard metacard) {
+        assertThat(metacard.getMetacardType()
+                .getName(), is(GmdMetacardType.GMD_METACARD_TYPE_NAME));
+    }
+
+    @Before
+    public void setUp() {
+        gmdTransformer = new GmdTransformer();
+    }
+
+    @Test(expected = IOException.class)
+    public void testBadInputStream() throws Exception {
+        InputStream is = Mockito.mock(InputStream.class);
+        doThrow(new IOException()).when(is)
+                .read(any());
+        gmdTransformer.transform(is);
+    }
+
+    @Test
+    public void testDataset() throws Exception {
+        Metacard metacard = transform("/gmd/dataset.xml");
+        assertGmdMetacard(metacard);
+        DateTimeFormatter dateFormatter = ISODateTimeFormat.dateOptionalTimeParser();
+        Date expectedDate = dateFormatter.parseDateTime("2004-03-14")
+                .toDate();
+
+        assertThat(metacard.getId(), is("550e8400-e29b-41d4-a716-48957441234"));
+
+        assertThat(metacard.getContentTypeName(), is("dataset"));
+
+        assertThat(metacard.getAttribute(GmdMetacardType.GMD_TYPE)
+                .getValue(), is("dataset"));
+
+        assertThat(metacard.getModifiedDate(), is(expectedDate));
+        assertThat(metacard.getCreatedDate(), is(expectedDate));
+
+        assertThat(metacard.getAttribute(GmdMetacardType.GMD_CRS)
+                .getValue(), is("urn:ogc:def:crs:World Geodetic System::WGS 84"));
+
+        assertThat(metacard.getTitle(), is("VMAPLV0"));
+        assertThat(metacard.getAttribute(GmdMetacardType.GMD_ABSTRACT)
+                .getValue(), is(
+                "Vector Map: a general purpose database design to support GIS applications"));
+
+        TreeSet<String> subjectAttributes = new TreeSet<>();
+        metacard.getAttribute(GmdMetacardType.GMD_SUBJECT)
+                .getValues()
+                .forEach(subject -> subjectAttributes.add((String) subject));
+        assertThat(SUBJECTS, is(subjectAttributes));
+
+        assertThat(metacard.getResourceURI()
+                .toASCIIString(), is("http://example.com"));
+        assertThat(metacard.getLocation(), is(
+                "POLYGON ((6.9 -44.94, 6.9 61.61, 70.35 61.61, 70.35 -44.94, 6.9 -44.94))"));
+
+    }
+
+    @Test
+    public void testFormat() throws Exception {
+        Metacard metacard = transform("/gmd/dataset2.xml");
+        assertGmdMetacard(metacard);
+
+        assertThat(metacard.getAttribute(GmdMetacardType.GMD_FORMAT)
+                .getValue(), is("shapefile"));
+    }
+
+    @Test
+    public void testAllMetadata() throws Exception {
+        InputStream input = getClass().getResourceAsStream("/gmd/");
+        BufferedReader rdr = new BufferedReader(new InputStreamReader(input));
```

Comment:
Not even really optional; I'm pretty sure this would fail a FindBugs scan because there's no guarantee the stream will be closed.